### PR TITLE
Fix android device-level "back" button closing the game

### PR DESCRIPTION
**Pull request by miterosan**

### CODE REVIEW

**Comment by swoolcock** [comment]
Is this firing the escape key?

**Comment by ojh050118** [comment]
> Is this firing the escape key?

Yes. It is recognized as an escape key.

File: osu.Framework.Android/AndroidGameActivity.cs [hunk 1]
```diff
@@ -32,5 +33,36 @@ protected override void OnCreate(Bundle savedInstanceState)
             // Because Android is not playing nice with Background - we just kill it
             System.Diagnostics.Process.GetCurrentProcess().Kill();
         }
+
+        /// <summary>
+        /// Avoid the default implementation that does close the app.
```
Comment (peppy) [inline osu.Framework.Android/AndroidGameActivity.cs]: this should be an inline comment, not xmldoc

File: osu.Framework.Android/AndroidGameActivity.cs [hunk 2]
```diff
@@ -32,5 +33,36 @@ protected override void OnCreate(Bundle savedInstanceState)
...
+        /// </summary>
+        public override void OnBackPressed()
+        {
+        }
+
+        /// <summary>
+        /// There is the rare situation that the view does not get the events. So the view just gets triggered directly.
```
Comment (peppy) [inline osu.Framework.Android/AndroidGameActivity.cs]: move to inline comment and explain what the "rare situation" is, rather than being vague.

**Comment by smoogipoo** [comment]
This seems ok to me. Won't push the button since it needs testing.

File: osu.Framework.Android/AndroidGameActivity.cs [hunk 3]
```diff
@@ -32,5 +33,29 @@ protected override void OnCreate(Bundle savedInstanceState)
...
+            // Avoid the default implementation that does close the app.
+            // This only happens when the back button could not be captured from OnKeyDown.
+        }
+
+        // On some devices and keyboard combinations the OnKeyDown event does not propagate the key event to the view.
+        // Here it is done manually to ensure that the keys actually land in the view.
+
+        public override bool OnKeyDown([GeneratedEnum] Keycode keyCode, KeyEvent e)
+        {
+            return gameView.OnKeyDown(keyCode, e);
```
Comment (peppy) [inline osu.Framework.Android/AndroidGameActivity.cs]: need to ensure this doesn't result in double-operations under any circumstance.

**Comment by peppy** [comment]
Just going to get this one in for now. Easier to merge and wait for potential issues to occur than to test android for now.

**Pull request merged**